src/identity/identity.ts

```diff
@@ -0,0 +1,355 @@
+   */
+  public async grantWriteAccess(identity: string): Promise<void> {
+    await this.ensureOwnerAndIdentityProfile();
+
+    const {
+      activeIdentity,
+      contractLoader,
+      did,
+      executor,
+      nameResolver,
+      underlyingAccount,
+    } = this.options;
+
+    // get the did for the current activeIdentity
+    const activeDidAddress = await did.convertIdentityToDid(activeIdentity);
+    const didDocumentToUpdate = await did.getDidDocument(activeDidAddress);
+    // add the new identity to the did document
+    didDocumentToUpdate.publicKey.push({
+      id: `${didDocumentToUpdate.id}#${identity}`,
+      type: 'Secp256k1VerificationKey2018',
+      controller: didDocumentToUpdate.id,
+      ethereumAddress: identity,
+    });
+    didDocumentToUpdate.authentication.push(`${didDocumentToUpdate.id}#${identity}`);
+    await did.setDidDocument(activeDidAddress, didDocumentToUpdate);
+
+    // apply the identity as key to the keyHolder contract
+    const keyHolderContract = await contractLoader.loadContract('KeyHolder', activeIdentity);
+    await executor.executeContractTransaction(
+      keyHolderContract,
+      'addKey',
+      { from: underlyingAccount },
+      nameResolver.soliditySha3(identity),
+      1,
+      1,
+    );
+  }
+
+  /**
+   * Removes the access, to act on behalf of the activeIdentity, for another identity. When removing
+   * read access, a bmail is sent, so the identity gets a notification with a attachment, which
+   * permisssion was removed.
+   *
+   * @param      {string}                identity      identity to remove the access for
+   * @param      {string}                type          read, write, readWrite
+   * @param      {IdentityBMailContent}  bmailContent  optional bmail content to inform the identity
+   */
+  public async removeAccess(
+    identity: string,
+    type: string,
+    bmailContent?: IdentityBMailContent,
+  ): Promise<void> {
+    const {
+      activeIdentity,
+      mailbox,
+      profile,
+    } = this.options;
+    // prevent old profiles and ensure, that only the owner is running this function
+    await this.ensureOwnerAndIdentityProfile();
+
+    // prevent wrong sharing types
+    if (type !== 'read' && type !== 'write' && type !== 'readWrite') {
+      throw new Error(`Unknown access type passed to "identity.removeAccess(${identity}, ${type})"`);
+    }
+
+    if (type.startsWith('read')) {
+      // save information into the current profile about the grant status
+      const addressBook = await profile.getAddressBook();
+      if (addressBook.profile[identity]) {
+        delete addressBook.profile[identity].hasIdentityAccess;
+        delete addressBook.profile[identity].identityAccessGranted;
+        delete addressBook.profile[identity].identityAccessNote;
+      }
+    } else {
+      // save information into the current profile about the grant status
+      await Promise.all([
+        profile.addProfileKey(identity, 'hasIdentityAccess', type),
```

Comment:
Is that right? If the removeAccess type is 'write', the identity gets granted write permissions here, or am I confused?

Reply:
```suggestion
        profile.addProfileKey(identity, 'hasIdentityAccess', 'read'),
```